features/dd-sdk-android-rum/src/test/kotlin/com/datadog/android/rum/internal/metric/slowframes/DefaultSlowFramesListenerTest.kt

```diff
@@ -307,44 +309,58 @@ internal class DefaultSlowFramesListenerTest {
             .isEqualTo(expectedTotalFrameDuration)
 
         assertThat(report.slowFramesRate(viewCreatedTimestampNs + viewDurationNs))
-            .isEqualTo(expectedSlowFramesDuration.toDouble() / expectedTotalFrameDuration)
+            .isEqualTo(expectedSlowFrameRate)
     }
 
     @Test
-    fun `M return 0 W slowFramesRate { totalFramesDurationNs = 0 }`(
-        @LongForgery(min = 1, max = MAX_DURATION_NS) viewDurationNs: Long
+    fun `M return 0 W slowFramesRate W resolveReport { totalFramesDurationNs = 0 }`(
+        @LongForgery(min = 1, max = MAX_DURATION_NS) viewDurationNs: Long,
+        forge: Forge
     ) {
         // Given
         testedListener.onViewCreated(viewId, viewCreatedTimestampNs)
+        testedListener.onFrame(forge.aFrameData(frameDurationUiNanos = 0))
 
         // When
         val report = testedListener.resolveReport(viewId)
 
         // Then
-        assertDoesNotThrow { // No ArithmeticException
-            assertThat(report.slowFramesDurationNs)
-                .isZero()
+        assertThat(report).isNotNull
+        assertThat(report!!.slowFramesRate(viewDurationNs)).isZero()
+    }
+
+    @Test
+    fun `M return 0 W freezeFramesRate W resolveReport { totalFramesDurationNs = 0 }`(
+        @LongForgery(min = 1, max = MAX_DURATION_NS) viewDurationNs: Long,
+        forge: Forge
+    ) {
+        // Given
+        testedListener.onViewCreated(viewId, viewCreatedTimestampNs)
+        testedListener.onFrame(forge.aFrameData(frameDurationUiNanos = 0))
 
-            assertThat(report.totalFramesDurationNs)
-                .isZero()
+        // When
+        val report = testedListener.resolveReport(viewId)
 
-            assertThat(report.slowFramesRate(viewCreatedTimestampNs + viewDurationNs)).isZero()
-        }
+        // Then
+        assertThat(report).isNotNull
+        assertThat(report!!.freezeFramesRate(viewDurationNs)).isZero()
```

Comment:
minor: also can be like the following

```suggestion
        assertThat(checkNotNull(report).freezeFramesRate(viewDurationNs)).isZero()
```

same above